### PR TITLE
Adjusting for upstream HAL API updates, needs more work.

### PR DESCRIPTION
**Pull request by computer-whisperer**
This is not ready to merge yet, I just needed a place to discuss stuff with the group.

Also, feel free to double check any of my work.

### CODE REVIEW

File: hal-base/hal/functions.py [hunk 1]
```diff
@@ -494,14 +484,14 @@ def cleanNotifier(notifier):
 #############################################################################
 # PDP
 #############################################################################
-getPDPTemperature = _STATUSFUNC("getPDPTemperature", C.c_double)
```
Comment (computer-whisperer) [inline hal-base/hal/functions.py]: An Issue I am running into is that on some functions, like these PDP functions, the "status" argument is not the final argument in the c++ API, like we expect. I suspect someone got lazy when they added arguments to these, and didn't rearrange them to match the rest of the API. See https://github.com/robotpy/allwpilib/blob/master/hal/include/HAL/PDP.hpp

Comment (virtuald) [inline hal-base/hal/functions.py]: We should get them to fix that, I don't want to do any extra work.

**Comment by computer-whisperer** [comment]
Also @PeterJohnson fyi the folders at http://www.tortall.net/~robotpy/hal/ seem to all be empty for builds newer than jenkins-stable-2015.340-112-g4e46692.

File: hal-base/hal/functions.py (hunk 1 above)
Comment (virtuald) [inline hal-base/hal/functions.py]: I've pushed a [changeset](https://usfirst.collab.net/gerrit/#/c/1085/) into gerrit to get this fixed, hopefully should get merged pretty quickly.

Comment (virtuald) [inline hal-base/hal/functions.py]: See the [halfix](https://github.com/robotpy/allwpilib/blob/halfix/hal/include/HAL/PDP.hpp) branch in allwpilib.

Comment (computer-whisperer) [inline hal-base/hal/functions.py]: Don't you need to fix the callers of these functions in both the java and c++ wpilib? Not just the HAL api?

Comment (virtuald) [inline hal-base/hal/functions.py]: Yes, look at the rest of the commit. ;)

Comment (computer-whisperer) [inline hal-base/hal/functions.py]: Ok, my bad.

**Comment by computer-whisperer** [comment]
I was able to run this successfully in both the simulation environment and a real roborio. Unless you see things that need to be changed, I think this is ready to merge. Only caveat is that the PDP functions will be broken until Dustin's change gets merged upstream.

**Comment by PeterJohnson** [comment]
Builds should be working again as of 2015.340-176.  Note I've changed the build to use gcc 4.8.2 so it will work out of the box with the 2016 image, but this may break things on the 2015 image.

**Comment by computer-whisperer** [comment]
Thanks!

**Pull request merged**